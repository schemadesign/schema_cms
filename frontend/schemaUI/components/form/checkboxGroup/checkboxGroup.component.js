import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import CheckboxGroupContext from './checkboxGroup.context';
import { getStyles } from './checkboxGroup.styles';
import { CheckboxOnIcon } from '../../icons/checkboxOnIcon';
import { CheckboxOffIcon } from '../../icons/checkboxOffIcon';
import { withStyles } from '../../styles/withStyles';

export class CheckboxGroupComponent extends PureComponent {
  static propTypes = {
    customStyles: PropTypes.object,
    customCheckboxStyles: PropTypes.object,
    isEdit: PropTypes.bool,
    value: PropTypes.any,
    onChange: PropTypes.func.isRequired,
    name: PropTypes.string.isRequired,
    children: PropTypes.array.isRequired,
    checkedIcon: PropTypes.any,
  };

  static defaultProps = {
    customStyles: {},
    customCheckboxStyles: {},
    checkedIcon: <CheckboxOnIcon />,
    uncCheckedIcon: <CheckboxOffIcon />,
  };

  render() {
    const {
      name,
      children,
      onChange,
      value,
      customStyles,
      isEdit,
      checkedIcon,
      uncCheckedIcon,
      customCheckboxStyles,
      theme,
    } = this.props;
    const context = { name, onChange, value, isEdit, checkedIcon, uncCheckedIcon, customCheckboxStyles };
    const { containerStyles } = getStyles(theme);
    const styles = { ...containerStyles, ...customStyles };

    return (
      <div style={styles}>
        <CheckboxGroupContext.Provider value={context}>{children}</CheckboxGroupContext.Provider>
      </div>
    );
  }
}

export const CheckboxGroup = withStyles(CheckboxGroupComponent);
