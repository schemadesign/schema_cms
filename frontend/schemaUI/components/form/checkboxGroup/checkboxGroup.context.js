import { createContext } from 'react';

const CheckboxGroupContext = createContext({});

export default CheckboxGroupContext;
