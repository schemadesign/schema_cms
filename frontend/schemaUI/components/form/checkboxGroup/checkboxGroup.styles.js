import { light } from '../../../utils/theme';

export const getStyles = (theme = light) => ({
  display: 'flex',
  flexDirection: 'column',
  borderTop: `1px solid ${theme.checkbox.border}`,
});
