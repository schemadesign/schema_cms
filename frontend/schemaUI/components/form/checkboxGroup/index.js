export { CheckboxGroup } from './checkboxGroup.component';
