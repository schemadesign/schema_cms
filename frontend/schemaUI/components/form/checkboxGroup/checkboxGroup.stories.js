import React from 'react';
import { storiesOf } from '@storybook/react';

import { CheckboxGroup } from './checkboxGroup.component';

const defaultProps = {};

storiesOf('CheckboxGroup', module).add('Default', () => <CheckboxGroup {...defaultProps} />);
