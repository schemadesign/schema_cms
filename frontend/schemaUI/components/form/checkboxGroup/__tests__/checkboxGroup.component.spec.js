import React from 'react';
import { shallow } from 'enzyme';

import { CheckboxGroup } from '../checkboxGroup.component';

describe('CheckboxGroup: Component', () => {
  const defaultProps = {};

  const component = props => <CheckboxGroup {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });
});
