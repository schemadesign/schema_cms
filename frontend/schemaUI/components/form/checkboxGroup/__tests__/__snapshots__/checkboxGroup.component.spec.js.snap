// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup: Component should render correctly 1`] = `
<div
  style={
    Object {
      "borderTop": "1px solid #F0EFEF",
      "display": "flex",
      "flexDirection": "column",
      "width": "320px",
    }
  }
>
  <ContextProvider
    value={
      Object {
        "checkedIcon": <CheckboxOnIcon />,
        "customCheckboxStyles": Object {},
        "isEdit": undefined,
        "name": "name",
        "onChange": [Function],
        "uncCheckedIcon": <CheckboxOffIcon />,
        "value": Array [
          "checkbox 1",
        ],
      }
    }
  >
    <Checkbox
      id="checkbox 1"
      key="1"
      value="checkbox 1"
    >
      checkbox 1
    </Checkbox>
    <Checkbox
      id="checkbox 2"
      isEdit={true}
      key="2"
      value="checkbox 2"
    >
      checkbox 2
    </Checkbox>
  </ContextProvider>
</div>
`;
